src/Oracles/PoolOracle.jl

```diff
@@ -6,6 +6,6 @@ function PoolOracle(data, labels, params::Dict{Symbol, Any})
     return PoolOracle(labels)
 end
 
-function ask_oracle(oracle::PoolOracle, query_id::Int)
-    return oracle.labels[query_id]
+function ask_oracle(oracle::PoolOracle, query_ids::Vector{Int})::Vector{Symbol}
```

Comment:
Why do we need to specify query_id as a vector? I'd suggest to rather leave specialization to the compiler. 
For the return type: I would also not specify this since the function is type stable (labels in PoolOracle already is type-annotated.)